### ISSUE TITLE
[3.6.0][CMake] Could NOT find XXXX (missing: XXXX_DIR) (Native Plugins, Android)

### ISSUE THREAD

**Issue by patrick-fu** [cocos/cocos-engine#12482]
### Cocos Creator version

3.6.0

### System information

macOS Editor (Mac with M1 chip, macOS 12.5), build for Android

### Issue description

I'm trying the [Native Plugins](https://docs.cocos.com/creator/manual/en/advanced-topics/native-plugins/brief.html) feature launched in Cocos Creator v3.6.0, and playing the demo https://github.com/PatriceJiang/ccplugin_tutorial

I have try to build for iOS / macOS and they works well

I have solve the problem that node can not be found by Android Studio ( #12456 )

And then I encounter next issue: cmake's "find_package()" can not find the plugin's XXXX-config.cmake

```cmake
# Pre-AutoLoadPlulgins.cmake

# plugins found & enabled in search path

set(hello_cocos_glue_ROOT
   "${CC_PROJECT_DIR}/../../plugins/hello_cocos/android/${ANDROID_ABI}"
   "${CC_PROJECT_DIR}/../../plugins/hello_cocos/android"
)

list(APPEND CMAKE_FIND_ROOT_PATH ${hello_cocos_glue_ROOT})
list(APPEND CC_REGISTERED_PLUGINS
  hello_cocos_glue
)

find_package(hello_cocos_glue
  NAMES "hello_cocos_glue"
  PATHS
    "${CC_PROJECT_DIR}/../../plugins/hello_cocos/android/${ANDROID_ABI}"
    "${CC_PROJECT_DIR}/../../plugins/hello_cocos/android"
  NO_DEFAULT_PATH
)
```

<img width="390" alt="image" src="https://user-images.githubusercontent.com/20736775/185566727-cfce806a-ab56-4c18-bfc0-54943627610b.png">

I've read the cmake documentation of "find_package" [https://cmake.org/cmake/help/latest/command/find_package.html#id4](https://cmake.org/cmake/help/latest/command/find_package.html#id4), the cmake code above (Pre-AutoLoadPlulgins.cmake) seems to be fine

I try to use newer version cmake to run it, (both 3.18.1 and 3.22.1 installed by Android Studio SDK manager) but it still not work (by setting cmake version in build.gradle -> externalNativeBuild, and it also needs to upgrade AGP to v4.2.0 and gradle to v6.7.1)

Then I try to run cmake command in a stand alone terminal (use the build arguments in `NewProject/build/android/proj/build/cmake/debug/arm64-v8a/build_command.txt`) and it success (both cmake 3.10.2.xxx / 3.18.1)

So I wonder if this is also an Android Studio environment related issue? And how to resolve this issue?

### Relevant error log output

platform: Android
Ignore NO_WERROR
OPTION USE_MODULES:	OFF
OPTION CC_USE_METAL:	
OPTION CC_USE_GLES3:	ON
OPTION CC_USE_GLES2:	ON
OPTION CC_USE_VULKAN:	OFF
OPTION CC_DEBUG_FORCE:	OFF
OPTION USE_SE_V8:	ON
OPTION USE_V8_DEBUGGER:	ON
OPTION USE_V8_DEBUGGER_FORCE:	OFF
OPTION USE_SE_SM:	OFF
OPTION USE_SOCKET:	ON
OPTION USE_AUDIO:	ON
OPTION USE_EDIT_BOX:	ON
OPTION USE_VIDEO:	ON
OPTION USE_WEBVIEW:	ON
OPTION USE_MIDDLEWARE:	ON
OPTION USE_DRAGONBONES:	ON
OPTION USE_SPINE:	ON
OPTION USE_WEBSOCKET_SERVER:	OFF
OPTION USE_PHYSICS_PHYSX:	OFF
OPTION USE_JOB_SYSTEM_TBB:	OFF
OPTION USE_JOB_SYSTEM_TASKFLOW:	OFF
OPTION USE_SERVER_MODE:	OFF
OPTION USE_CCACHE:	OFF
OPTION CCACHE_EXECUTABLE:	/opt/homebrew/bin/ccache
OPTION NODE_EXECUTABLE:	/Users/patrickfu/.nvm/versions/node/v16.14.0/bin/node
OPTION NET_MODE:	0
OPTION USE_REMOTE_LOG:	OFF
platform path: 
Try generating /Users/patrickfu/code/cocos/ccplugin_tutorial/NewProject/native/engine/android/Pre-AutoLoadPlulgins.cmake
 execute plugin_parser.js 
Searching hook files Pre*.cmake or *Pre.cmake in /Users/patrickfu/code/cocos/ccplugin_tutorial/NewProject/native/engine/android
::Loading Pre-AutoLoadPlulgins.cmake
Could NOT find hello_cocos_glue (missing: hello_cocos_glue_DIR)
::Loading Pre-service.cmake
Generating plugin_registry code, total 1 plugins
Searching hook files Post*.cmake or *Post.cmake in /Users/patrickfu/code/cocos/ccplugin_tutorial/NewProject/native/engine/android
::Loading Post-service.cmake
Configuring done
CMake Deprecation Warning:
  The 'cmake-server(7)' is deprecated.  Please port clients to use the
  'cmake-file-api(7)' instead.

### Steps to reproduce

1. Clone https://github.com/PatriceJiang/ccplugin_tutorial
2. Open the project with Cocos Creator v3.6.0
3. Build for Android (export the android project)
4. Open the exported android project, then run gradle sync
5. See the "build_output.txt"

### Minimal reproduction project

https://github.com/PatriceJiang/ccplugin_tutorial

**Comment by PatriceJiang** [comment]
@patrick-fu Can you try this fix https://github.com/cocos/cocos-engine/pull/12550

I'm not sure at the moment why `<PackageName>_ROOT` doesn't work on Android

**Comment by patrick-fu** [comment]
@PatriceJiang #12550 works fine! 👍    Will this fix be released in version 3.6.1?

**Comment by PatriceJiang** [comment]
Yes
